zetta_utils/mazepa_layer_processing/common/subchunkable_apply_flow.py

```diff
@@ -473,8 +484,10 @@ def _expand_bbox_processing(  # pylint: disable=line-too-long
             f"{dst_resolution.pformat()}. You may set `expand_bbox_resolution = True` to "
             "automatically expand the bbox to the nearest integral pixel."
         )
-    translation_end = (processing_chunk_sizes[0] - bbox_shape_in_res) % processing_chunk_sizes[0]
     bbox_old = bbox
+    chunk_size = processing_chunk_sizes[0]
+    # Round up the bbox so that (bbox - chunk size) is an even multiple of (chunk size + gap).
+    translation_end = (chunk_size - bbox_shape_in_res) % (chunk_size + gap)
```

Comment:
Minor, but `chunk_size` is somewhat ambiguous - better to call this `chunk_size_top`.